hibernate-core/src/main/java/org/hibernate/dialect/MySQLDialect.java

```diff
@@ -137,22 +138,29 @@ public Size resolveSize(
 	private final int maxVarcharLength;
 	private final int maxVarbinaryLength;
 
+	private final boolean noBackslashEscapesEnabled;
+
 	public MySQLDialect() {
 		this( MINIMUM_VERSION );
 	}
 
 	public MySQLDialect(DatabaseVersion version) {
-		this( version, 4 );
+		this( version, 4, false );
 	}
 
-	public MySQLDialect(DatabaseVersion version, int bytesPerCharacter) {
+	public MySQLDialect(DatabaseVersion version, int bytesPerCharacter, boolean noBackslashEscapes) {
```

Comment:
First, we need a new constructor and leave the signature of the old one untouched for backwards compatibility. Apart from that, I'd like to ask you to introduce a `MySQLServerConfiguration` class, which encapsulates the `bytesPerCharacter` and `noBackslashEscapes`, which we determine in a single query through a factory method `MySQLServerConfiguration.fromDatabaseMetadata(DatabaseMetadata)`. The legacy dialects can then also use that.

Reply:
Should I leave the protected method `getCharacterSetBytesPerCharacter` in `MySQLDialect` for compatibility?

Reply:
Yeah, but let's deprecate them.